src/wordlift/mappings/class-mappings-dbo.php

```diff
@@ -55,27 +55,37 @@ public function get_mappings() {
 	public function get_active_mappings() {
 		$mapping_table_name = $this->wpdb->prefix . WL_MAPPING_TABLE_NAME;
 
-		return $this->wpdb->get_results(
+		$mappings = $this->wpdb->get_results(
 			"SELECT * FROM $mapping_table_name WHERE mapping_status = 'active'",
 			ARRAY_A
 		);
+
+		return apply_filters( 'wl_mappings_register_mappings', $mappings );
+
 	}
 
 	/**
 	 * Returns a list of property rows
 	 *
-	 * @param int $mapping_id Primary key of mapping table.
+	 * @param string $mapping_id Primary key of mapping table.
 	 *
 	 * @return array List of property items belong to $mapping_id.
 	 */
 	public function get_properties( $mapping_id ) {
-		$property_table_name = $this->wpdb->prefix . WL_PROPERTY_TABLE_NAME;
-		$property_rows       = $this->wpdb->get_results(
-			$this->wpdb->prepare( "SELECT * FROM $property_table_name WHERE mapping_id=%d", $mapping_id ),
-			ARRAY_A
-		);
-
-		return $property_rows;
+		$property_table_name   = $this->wpdb->prefix . WL_PROPERTY_TABLE_NAME;
+		$ui_defined_properties = array();
+		if ( is_numeric($mapping_id) ) {
+			$ui_defined_properties = $this->wpdb->get_results(
+				$this->wpdb->prepare( "SELECT * FROM $property_table_name WHERE mapping_id=%d", (int)$mapping_id ),
+				ARRAY_A
+			);
+		}
+		$manual_properties = apply_filters('wl_mappings_register_properties', array());
```

Comment:
@naveen17797 pls 

- [ ] add PhpDoc for the filter and

- [ ] change the filter to provide the mapping's id in the parameters, so that hooks can provide additional properties based on that and you don't need to filter at ln 85. This is extremely important as potentially we may have 1.000s of properties. 

- [ ] provide as initial value `$ui_defined_properties` instead of `array()` so that hooks can also modify existing data and you can remove the `array_merge` at ln 88.